File: 0331-verify-preorder-serialization-of-a-binary-tree/0331-verify-preorder-serialization-of-a-binary-tree.cpp
class Solution {
public:
    bool isValidSerialization(string preorder) {
        stringstream ss(preorder);
        string element;
        int diff = 1;
        while(getline(ss,element,',')) {
            if(--diff<0) return false;
            if(element!="#")diff+=2;
        }
        return diff==0;
    }
};

File: 0331-verify-preorder-serialization-of-a-binary-tree/0331-verify-preorder-serialization-of-a-binary-tree.java
class Solution {
    public boolean isValidSerialization(String preorder) {
        String[] nodes = preorder.split(",");
        //difference of Edges( Outdegree - Indegree);
        //root special case diff=1;
        int diff = 1;
        for(String node : nodes) {
            if(--diff<0)return false;
            if(!node.equals("#")) diff+=2;
        }
        return diff==0;
    }
}